internal/controller/runtime/fsm/seed.go

```diff
@@ -0,0 +1,29 @@
+package fsm
+
+import (
+	"context"
+	gardener_types "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func seedForRegionAvailable(client client.Client, region string) (bool, []string, error) {
+	var seedList gardener_types.SeedList
+	var regionsWithSeeds []string
+
+	err := client.List(context.TODO(), &seedList)
```

Comment:
We can pass context from `sFnCreateShoot` function state here.

Reply:
Done.